docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,404 @@
+.. _custom_core:
+
+Adding a custom core
+====================
+
+You may want to add a custom RISC-V core to Chipyard generator. If the top module of your core is not in Chisel, 
+you will first need to create a Verilog blackbox for it. See :ref:`incorporating-verilog-blocks` for instructions.
+Once you have a top module in Chisel, you are ready to create integrate it with Chipyard. 
+
+.. note:: 
+
+    RoCC is not supported by custom core currently. Please use Rocket or Boom if you need to use RoCC.
+
+Parameter Case Classes
+----------------------
+
+Chipyard will generate a core for every ``TileParams`` object it discovered in the current config.
```

Comment:
I would like more details here on the TilesLocated field, which is the actual data structure that drives Tile elaboration

Reply:
Done.